### ISSUE TITLE
Share UX

### ISSUE THREAD

**Issue by erasche** [GMOD/Apollo#1122]
Just wanted to document this. I have some work in a branch but still feeling my way around GWT. It'll be a few months before I get something usable implemented (due to other priorities)

My users wish to control sharing of the genomes themselves. I realise not every site wants that (some may want the org admin, or the sys admin to control sharing + dissemination and vet access), but my site strongly would like google docs style sharing so they don't have to contact their admin to add a user to a specific genome.

Something like the following (but in english, obviously).

![utvalg_849](https://cloud.githubusercontent.com/assets/458683/16014167/936440fc-317f-11e6-88e1-25d1d3e6e854.png)

**Comment by nathandunn** [comment]
@erasche  I want to be able to make sure I understand the use-case.   You want to be able to configure it such that:

1 - non-admin users can invite other user's not in the system to edit
2 - non-admin users can invite other user's IN the system to edit a genome that they can also edit

What groups like @childers  did is build a system on top of Apollo to do this.   I think in their model user's requested access and on approval it added to Apollo via web-services (I hope that is right). 

I would advocate something similar, or you could build a system in-place to do the same.   I would probably just add an additional reports page (creating another tab that non-admin user's can see) and use that to send invites or build a system to interact with Apollo directly.   We can discuss this more.

**Comment by erasche** [comment]
@nathandunn 

> 1 - non-admin users can invite other user's not in the system to edit

No. I'm fine with users having to go through a registration process as of now.

> 2 - non-admin users can invite other user's IN the system to edit a genome that they can also edit

yes! This. 

> What groups like @childers did is build a system on top of Apollo to do this. I think in their model user's requested access and on approval it added to Apollo via web-services (I hope that is right). 

Building on top... I would do this but I'm not sure of the cost/benefit here. If I did, it would likely go all in, and build a full-featured AngularJS standalone client and just use Apollo's API. That's a lot of work for something that I feel would fit pretty naturally with the existing collaborative, real-time <s>document</s>annotation editing system.

> I would advocate something similar, or you could build a system in-place to do the same. I would probably just add an additional reports page (creating another tab that non-admin user's can see) and use that to send invites or build a system to interact with Apollo directly. We can discuss this more. 

To me, this is more attractive. My initial stab at it was to add a new tab which would allow the user to share the organism they're on with others in the system.

Anyway. This is something I'll probably work on more after GMOD conference. Long term thing.
